tests/integration/test_charm.py

```diff
@@ -60,10 +50,37 @@ async def test_deploy(
         CUSTOM_RESOURCE_GROUP_NAME, "v1", GATEWAY_RESOURCE_NAME, GATEWAY_PLURAL
     )
     gateway = lightkube_client.get(gateway_generic_resource_class, name=application.name)
-    assert len(gateway.spec["listeners"]) == 2
-    secret: Secret = lightkube_client.get(
-        Secret, name=f"{application.name}-secret-{TEST_EXTERNAL_HOSTNAME_CONFIG}"
+    gateway_lb_ip = gateway.status["addresses"][0]["value"]  # type: ignore
+    assert gateway_lb_ip, "LB address not assigned to gateway"
+
+    ingress_url = (
+        f"http://{gateway_lb_ip}/{application.model.name}-{ingress_requirer_application.name}"
+    )
```

Comment:
A better way to get that url is to add an action on AnyCharm that returns the URL provided in the relation, it allows you to test that the URL in the relation is also correct

Reply:
And since you want HTTP you can do an URL parse and change the scheme

Reply:
I wanted to use an actual charm with workload for this, so I added a helper function to get the ingress url from the unit data. I also agree with adding an action on AnyCharm and I'll add that in a follow up PR